nx run returned exit code 0 when tests are failing

### Current Behavior

When running `nx affected -t lint,test,build,package` my CI status checks pass even though tests are failing because the exit code is 0.

Doing this locally as `nx run example-service:test` also returns exit code 0

https://github.com/Voziv/monorepo-nx/actions/runs/7714883819/job/21028163809?pr=2#step:7:61

### Expected Behavior

NX should have a non-zero exit code when targets fail

### GitHub Repo

https://github.com/Voziv/monorepo-nx

### Steps to Reproduce

1. Clone https://github.com/Voziv/monorepo-nx
2. Check out the `add-new-test` branch, `git checkout add-new-test`
3. `npm i`
4. Run `nx run example-service:test; echo "NX exited with code: $?"`
5. See that the exit code was 0


### Nx Report

```shell
❯ nx report

 >  NX   Report complete - copy this into the issue template

   Node   : 20.10.0
   OS     : linux-x64
   npm    : 10.2.3
   
   nx (global)   : 17.2.2
   nx            : 17.3.0
   @nx/devkit    : 17.3.0
   @nrwl/devkit  : 17.2.7
   @nrwl/tao     : 17.3.0
   typescript    : 4.9.5
   ---------------------------------------
   Community plugins:
   @theunderscorer/nx-semantic-release : 2.10.0
   ---------------------------------------
   The following packages should match the installed version of nx
     - @nrwl/devkit@17.2.7
   
   To fix this, run `nx migrate nx@17.3.0`
```


### Failure Logs

```shell
~/dev/voziv/monorepo-nx add-new-test
❯ nx run example-service:test

> nx run example-service:test

> example-service@1.0.0 test
> jest
 FAIL  __tests__/handler.test.ts
  ✓ handler should return expected message (1 ms)
  ✕ nx should fail on  failure
  ● nx should fail on  failure
    ReferenceError: fail is not defined

      6 |
      7 | test('nx should fail on  failure', async () => {
    > 8 |     fail('PLEASE EXIT WITH NON ZERO')
        |     ^
      9 | });

      at __tests__/handler.test.ts:8:5
      at __tests__/handler.test.ts:8:71
      at Object.<anonymous>.__awaiter (__tests__/handler.test.ts:4:12)
      at Object.<anonymous> (__tests__/handler.test.ts:7:47)
Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 passed, 2 total
Snapshots:   0 total
Time:        0.428 s, estimated 1 s
Ran all test suites.
npm ERR! Lifecycle script `test` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: example-service@1.0.0 
npm ERR!   at location: /home/voziv/dev/voziv/monorepo-nx/services/example-service
```


### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

I ran `nx migrate` but `nx report` still shows `@nrwl/devkit@17.2.7`.

--- a/packages/nx/src/executors/run-script/run-script.impl.ts
+++ b/packages/nx/src/executors/run-script/run-script.impl.ts
@@ -1,10 +1,8 @@
-import { getPackageManagerCommand } from '../../utils/package-manager';
-import type { ExecutorContext } from '../../config/misc-interfaces';
 import * as path from 'path';
-import { env as appendLocalEnv } from 'npm-run-path';
+import type { ExecutorContext } from '../../config/misc-interfaces';
 import { runCommand } from '../../native';
 import { PseudoTtyProcess } from '../../utils/child-process';
-import { signalToCode } from '../../utils/exit-codes';
+import { getPackageManagerCommand } from '../../utils/package-manager';
 
 export interface RunScriptOptions {
   script: string;
@@ -25,10 +23,7 @@ export default async function (
             context.root,
             context.projectsConfigurations.projects[context.projectName].root
           ),
-          {
-            ...process.env,
-            ...appendLocalEnv(),
-          }
+          process.env
         )
       );
       cp.onExit((code) => {

